packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -0,0 +1,285 @@
+  ///
+  /// Typically a [Text] widget.
+  final Widget title;
+
+  /// The (optional) content of the dialog is displayed in the center of the
+  /// dialog in a lighter font.
+  ///
+  /// Typically a [Text] widget.
+  final Widget content;
+
+  /// The (optional) set of actions that are displayed at the bottom of the
+  /// dialog.
+  ///
+  /// Typically this is a list of [CupertinoDialogAction] widgets.
+  final List<Widget> actions;
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> children = <Widget>[];
+
+    children.add(new SizedBox(height: 20.0));
+
+    if (title != null) {
+      children.add(new Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 20.0),
+        child: new DefaultTextStyle(
+          style: _kCupertinoDialogTitleStyle,
+          textAlign: TextAlign.center,
+          child: title,
+        ),
+      ));
+    }
+
+    if (content != null) {
+      children.add(new Flexible(
+        fit: FlexFit.loose,
+        child: new Padding(
+          padding: const EdgeInsets.symmetric(horizontal: 20.0),
+          child: new DefaultTextStyle(
+            style: _kCupertinoDialogContentStyle,
+            textAlign: TextAlign.center,
+            child: content,
+          ),
+        ),
+      ));
+    }
+
+    children.add(new SizedBox(height: 20.0));
+
+    if (actions != null) {
+      children.add(new _CupertinoButtonBar(
+        children: actions,
+      ));
+    }
+
+    return new CupertinoDialog(
+      child: new IntrinsicWidth(
+        child: new Column(
+          mainAxisSize: MainAxisSize.min,
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: children,
+        ),
+      ),
+    );
+  }
+}
+
+const Color _kDestructiveActionColor = const Color(0xFFFF3B30);
+
+/// A button typically used in a [CupertinoAlertDialog].
+///
+/// See also:
+///
+///  * [CupertinoAlertDialog]
+class CupertinoDialogAction extends StatelessWidget {
+  /// Creates an action for an iOS-style dialog.
+  CupertinoDialogAction({
+    this.onPressed,
+    this.isDestructive: false,
+    this.child,
+  });
+
+  /// The callback that is called when the button is tapped or otherwise activated.
+  ///
+  /// If this is set to null, the button will be disabled.
+  final VoidCallback onPressed;
+
+  /// Whether this action destroys an object.
+  ///
+  /// For example, an action that deletes an email is destructive.
+  final bool isDestructive;
+
+  /// The widget below this widget in the tree.
+  ///
+  /// Typically a [Text] widget.
+  final Widget child;
+
+  /// Whether the button is enabled or disabled. Buttons are disabled by default. To
+  /// enable a button, set its [onPressed] property to a non-null value.
+  bool get enabled => onPressed != null;
+
+  @override
+  Widget build(BuildContext context) {
+    TextStyle style = _kCupertinoDialogActionStyle;
+
+    if (isDestructive)
+      style = style.copyWith(color: _kDestructiveActionColor);
+
+    if (enabled)
```

Comment:
we should have a test that CupertinoDialogAction picks the right colours and styles in the face of the various options